cubeviz/controls/flux_unit_registry.py

```diff
@@ -0,0 +1,289 @@
+        """
+        units = ["pixel"]
+
+        spaxel = u.def_unit('spaxel', u.astrophys.pix)
+        u.add_enabled_units(spaxel)
+        units.append('spaxel')
+
+        return units
+
+    def is_compatible(self, unit):
+        """
+        Check if unit can be converted to units
+        in this registry. Uses model units to check.
+        :param unit: Unit or unit str
+        :return: bool
+        """
+        compatible = False
+        for model_unit in self._model_unit:
+            try:
+                model_unit.to(unit)
+                compatible = True
+            except u.UnitConversionError:
+                continue
+        return compatible
+
+    def get_unit_list(self, pixel_only=False,
+                      solid_angle_only=False,
+                      current_unit=None):
+        """
+        Returns a list of unit strings in the registry.
+        Adds current_unit if not duplicated. Registered
+        units are added first.
+        :param pixel_only: (bool) Return pixel units only
+        :param solid_angle_only: (bool) Return solid_angle_only units only
+        :param current_unit: Unit or unit str
+        :return: list of unit str
+        """
+        unit_list = []
+        item_list = []
+        if not solid_angle_only:
+            unit_list.extend(self._locally_defined_pixel_units())
+            item_list.extend(self.runtime_pixel_units)
+        if not pixel_only:
+            unit_list.extend(self._locally_defined_solid_angle_units())
+            item_list.extend(self.runtime_solid_angle_units)
+
+        for item in item_list:
+            unit_list = _add_unit_to_list(unit_list, item)
+
+        if current_unit is not None:
+            if isinstance(current_unit, str):
+                current_unit = u.Unit(current_unit)
+            unit_list = _add_unit_to_list(unit_list, current_unit)
+        return unit_list
+
+    def add_pixel_unit(self, item):
+        """
+        Add new pixel unit
+        :param item: unit or unit str
+        """
+        if isinstance(item, str) \
+                or isinstance(item, u.UnitBase):
+            if item not in self.runtime_pixel_units:
+                self.runtime_pixel_units.append(item)
+
+    def add_solid_angle_unit(self, item):
+        """
+        Add new solid angle unit
+        :param item: unit or unit str
+        """
+        if isinstance(item, str) \
+                or isinstance(item, u.UnitBase):
+            if item not in self.runtime_solid_angle_units:
+                self.runtime_solid_angle_units.append(item)
+
+    def add_unit(self, item):
+        """
+        Check for type and add new unit
+        :param item: unit or unit str
+        """
+        if isinstance(item, str):
+            new_unit = u.unit(item)
+        else:
+            new_unit = item
+
+        if new_unit.decompose() == u.pix.decompose():
+            self.add_pixel_unit(new_unit)
+        if 'solid angle' in new_unit.physical_type:
```

Comment:
I haven't looked at the physical types but it might be safer to do `new_unit.physical_type.lower()`

Reply:
I will change this now.